### PR TITLE
Updated the PalindromeChecker java code

### PR DESCRIPTION
**Pull request by CodeXstalker**
Hey @ReciHub ,
I added the optimized java code for Palindrome Number, please have a look(#1028).

### CODE REVIEW

**Comment by CodeXstalker** [comment]
@ReciHub please have a look

File: PalindromeChecker/OptimizedPalindromeChecker.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//Fastest runninig algorithm for Plindrome checker and optimized too
+
+class Solution {
+public boolean isPalindrome(int x) {
```
Comment (SaiSudhaV) [inline PalindromeChecker/OptimizedPalindromeChecker.java:4]: This code works for numbers but the previous is for strings. Please do add a new file instead modifying as these 2 are different functionalities.

**Review by SaiSudhaV — CHANGES_REQUESTED** [review]

File: PalindromeChecker/OptimizedPalindromeChecker.java (hunk 1 above)
Comment (CodeXstalker) [reply PalindromeChecker/OptimizedPalindromeChecker.java:4]: @ReciHub , I request you to merge for the number, Coz it is best optimized for it.

**Comment by CodeXstalker** [comment]
@ReciHub I added the algorithm for Prime Checker using Regular Expression in java.
So, Have a look and consider  it.

**Review by SaiSudhaV — APPROVED** [review]

**Pull request merged**